docs/api/implementation-guide.md

```diff
@@ -0,0 +1,100 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2018, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## Chronology
+
+To assist authors of existing NuGet repositories keep up to date with NuGet's newest features, here is the chronology of the relevant features mentioned in the remainder of the document.
+
+|Version|Feature|
```

Comment:
I think this should say "NuGet client version" since we don't really have a protocol version aside from "V3" and some resource version strings which are not guaranteed to be exactly matching the respective client version that first shipped the feature.

I think a date column would be best for the first column. That would be easiest for people to remember or refer against their own effort (e.g. "oh yeah I first implemented my Go-based server in 2021 based on Git commits, so I should look at the stuff after 2021").